collector/lib/IntrospectionEndpoint.h

```diff
@@ -0,0 +1,27 @@
+#ifndef _INTROSPECTION_ENDPOINT_
+#define _INTROSPECTION_ENDPOINT_
+
+#include <CivetServer.h>
+#include <optional>
+#include <unordered_map>
+
+namespace collector {
+
+using QueryParams = std::unordered_map<std::string, std::string>;
+
+class IntrospectionEndpoint : public CivetHandler {
+ protected:
+  static QueryParams ParseParameters(const char* queryString);
+  static std::optional<std::string> GetParameter(const QueryParams& params, const std::string& paramName);
```

Comment:
These methods are not used in the PR. However, I think they will be soon, as we add more endpoints.